code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -110,12 +110,15 @@
 	if(!T || !istype(T))
 		return
 
-	if(!hive?.living_xeno_queen || hive.living_xeno_queen.loc.z == loc.z) //if there is a queen, it must be in the same z-level
-		if(locate(/obj/effect/alien/weeds) in T || xeno_caste.caste_flags & CASTE_INNATE_HEALING) //We regenerate on weeds or can on our own.
-			if(lying || resting)
-				heal_wounds(XENO_RESTING_HEAL)
-			else
-				heal_wounds(XENO_STANDING_HEAL) //Major healing nerf if standing.
+	var/queen_healing_penalty = 0.5
+	if(!hive || (hive.living_xeno_queen && hive.living_xeno_queen.loc.z == T.z)) //if we are hiveless or the living queen's z-level is the same as ours.
```

Comment:
```suggestion
	if(hive?.living_xeno_queen?.loc?.z == T.z)) //if the living queen's z-level is the same as ours.
```
no xeno should ever be hiveless.

Reply:
That is bad don't do checking if something exists with comparison. This way if the left-side is null and they are in nullspace it would pass.